### PR TITLE
feat: Add DartFrog JWT-Auth Example

### PR DESCRIPTION
**Pull request by codekeyz**
## Type of Change

- [ ] ✨ `feat` -- New feature (non-breaking change which adds functionality)
- [ ] 🛠️ `fix` -- Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ `!` -- Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 `refactor` -- Code refactor
- [ ] ✅ `ci` -- Build configuration change
- [ ] 📝 `docs` -- Documentation
- [ ] 🗑️ `chore` -- Chore

### CODE REVIEW

File: templates/crud_rest_api_dartfrog_jwt/lib/env.dart [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import 'package:envied/envied.dart';
+
+part 'env.g.dart';
```
Comment (lesnitsky) [inline templates/crud_rest_api_dartfrog_jwt/lib/env.dart:3]: could we drop envied and use dotenv instead? makes the whole setup a bit easier

**Review by lesnitsky — CHANGES_REQUESTED** [review]

File: templates/dartfrog_jwt_auth/TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+### REST Endpoints
+
+- Register User
+
+  ```shell
+  curl -X POST -H "Content-Type: application/json" -d '{"username": "john", "password": "password123"}' http://localhost:8080/register
+  ```
+
+  ```json
```
Comment (lesnitsky) [inline templates/dartfrog_jwt_auth/TEMPLATE.md:40]: below is not json :)

File: templates/dartfrog_jwt_auth/routes/me/_middleware.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+final class AuthData {
+  final String username;
+  const AuthData(this.username);
+}
+
+Handler middleware(Handler handler) {
+  return (context) async {
+    // Retrieve token from header eg: `Bearer xxxxxx` -> `xxxxxx`
+    final bearerToken = context.request.headers[HttpHeaders.authorizationHeader]
```
Comment (lesnitsky) [inline templates/dartfrog_jwt_auth/routes/me/_middleware.dart:16]: can we have this logic aligned with https://github.com/invertase/globe/pull/73/files#diff-38f5d3106afa19697950a7c86e88f031767c7df053f59ae5f1e64a014ace49d4

**Review by lesnitsky — CHANGES_REQUESTED** [review]

File: templates/dartfrog_jwt_auth/routes/register.dart [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  if (fauxUserDB.containsKey(reqBody.username)) {
+    return Response(
+      statusCode: HttpStatus.badRequest,
+      body: 'Username already taken.',
+    );
+  }
+
+  final String hashedPassword =
```
Comment (lesnitsky) [inline templates/dartfrog_jwt_auth/routes/register.dart:42]: do we need a type here?

File: templates/dartfrog_jwt_auth/TEMPLATE.md (hunk 2 above)
Comment (codekeyz) [reply templates/dartfrog_jwt_auth/TEMPLATE.md]: Resolved.

File: templates/dartfrog_jwt_auth/routes/me/_middleware.dart (hunk 3 above)
Comment (codekeyz) [reply templates/dartfrog_jwt_auth/routes/me/_middleware.dart]: Resolved.

File: templates/dartfrog_jwt_auth/routes/login.dart [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        ),
+      _ => throw FormatException('Username & Password required for register'),
+    };
+  } on FormatException catch (e) {
+    return Response(statusCode: HttpStatus.badRequest, body: e.message);
+  }
+
+  final userFromDB = fauxUserDB[reqBody.username];
+  if (userFromDB == null ||
+      !BCrypt.checkpw(reqBody.password, userFromDB.passwordHash)) {
```
[the remark this answers is not in the document]
Comment (codekeyz) [reply templates/dartfrog_jwt_auth/routes/login.dart]: Resolved.

File: templates/dartfrog_jwt_auth/routes/register.dart (hunk 4 above)
Comment (codekeyz) [reply templates/dartfrog_jwt_auth/routes/register.dart:42]: Nope.

**Review by lesnitsky — APPROVED** [review]

**Pull request merged**